pkg/events/event.go

```diff
@@ -0,0 +1,73 @@
+package events
+
+import (
+	"time"
+
+	"github.com/litmuschaos/chaos-runner/pkg/utils"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientTypes "k8s.io/apimachinery/pkg/types"
+)
+
+//CreateEvents create the events in the desired resource
+func CreateEvents(eventsDetails *utils.EventDetails, clients utils.ClientSets) error {
+
+	events := &apiv1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      eventsDetails.Reason + string(eventsDetails.ResourceUID),
+			Namespace: eventsDetails.Namespace,
+		},
+		Source: apiv1.EventSource{
+			Component: eventsDetails.ResourceName + "-runner",
+		},
+		Message:        eventsDetails.Message,
+		Reason:         eventsDetails.Reason,
+		Type:           eventsDetails.Type,
+		Count:          1,
+		FirstTimestamp: metav1.Time{Time: time.Now()},
+		LastTimestamp:  metav1.Time{Time: time.Now()},
+		InvolvedObject: apiv1.ObjectReference{
+			APIVersion: "litmuschaos.io/v1alpha1",
+			Kind:       "ChaosEngine",
+			Name:       eventsDetails.ResourceName,
+			Namespace:  eventsDetails.Namespace,
+			UID:        eventsDetails.ResourceUID,
+		},
+	}
+
+	_, err := clients.KubeClient.CoreV1().Events(eventsDetails.Namespace).Create(events)
+	return err
+
+}
+
+//GenerateEvents update the events and increase the count by 1, if already present
+// else it will create a new event
+func GenerateEvents(eventsDetails *utils.EventDetails, clients utils.ClientSets) error {
+
+	var err error
+	event, _ := clients.KubeClient.CoreV1().Events(eventsDetails.Namespace).Get(eventsDetails.Reason+string(eventsDetails.ResourceUID), metav1.GetOptions{})
+
+	if event.Name != eventsDetails.Reason+string(eventsDetails.ResourceUID) {
+
+		err = CreateEvents(eventsDetails, clients)
+
+	} else {
+
+		event.LastTimestamp = metav1.Time{Time: time.Now()}
+		event.Count = event.Count + 1
+
+		_, err = clients.KubeClient.CoreV1().Events(eventsDetails.Namespace).Update(event)
+	}
+	return err
+}
+
+//SetEngineEventAttributes initialise attributes for event generation in chaos engine
+func SetEngineEventAttributes(eventsDetails *utils.EventDetails, Reason, Message, Type string, engineDetails *utils.EngineDetails) {
```

Comment:
Reduced the no. of arguments. We are copying three attributes now. I am avoiding builder for this as this will not be a reusable builder and we need three extra functions for this.